infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/ILogReplicationVersionAdapter.java

```diff
@@ -12,4 +14,11 @@ public interface ILogReplicationVersionAdapter {
      * @return Version string that indicates the version of LR.
      */
     String getVersion();
+
+    /**
+     * @return true if the cluster is undergoing rolling upgrades with
+     * some nodes running an older code base.
+     * false if all the nodes in the cluster are at the same version as me.
+     */
+    boolean isUpgradeRunning(TxnContext txnContext);
```

Comment:
would it be clearer to rename this method to something like 'isRollingUpgradeInProgress' or 'isUpgradeInProgress'?

Reply:
sure, changed.